engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/compensation/CompensationEventExecutionTest.java

```diff
@@ -130,4 +133,98 @@ public void shouldExecuteAProcessWithCompensationEndEvent() {
                 ProcessInstanceIntent.ELEMENT_COMPLETED,
                 BpmnEventType.UNSPECIFIED));
   }
+
+  @Test
+  public void shouldCreateAndUpdateCompensationSubscriptionForCompletedTask() {
+    // given
+    final var process =
+        createModelFromClasspathResource("/compensation/compensation-throw-event.bpmn");
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    ENGINE.job().ofInstance(processInstanceKey).withType(Protocol.USER_TASK_JOB_TYPE).complete();
+
+    // then
+    assertThat(
+            RecordingExporter.compensationSubscriptionRecords()
+                .withProcessInstanceKey(processInstanceKey))
+        .extracting(Record::getValueType, Record::getIntent)
+        .containsSequence(
+            tuple(ValueType.COMPENSATION_SUBSCRIPTION, CompensationSubscriptionIntent.CREATED),
+            tuple(ValueType.COMPENSATION_SUBSCRIPTION, CompensationSubscriptionIntent.TRIGGERED));
+  }
+
+  @Test
+  public void shouldActivateCompensationHandlerForIntermediateThrowEvent() {
+    // given
+    final var process =
+        createModelFromClasspathResource("/compensation/compensation-throw-event.bpmn");
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    ENGINE.job().ofInstance(processInstanceKey).withType(Protocol.USER_TASK_JOB_TYPE).complete();
+
+    // then
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limitToProcessInstanceCompleted())
+        .extracting(
+            r -> r.getValue().getBpmnElementType(),
+            Record::getIntent,
+            r -> r.getValue().getElementId())
+        .containsSubsequence(
+            tuple(
+                BpmnElementType.USER_TASK,
+                ProcessInstanceIntent.ELEMENT_ACTIVATING,
+                "Activity_1epoz0g"),
+            tuple(
+                BpmnElementType.USER_TASK,
+                ProcessInstanceIntent.ELEMENT_ACTIVATED,
+                "Activity_1epoz0g"));
```

Comment:
:wrench: We could verify the sequence of the BPMN elements: user task, compensation throw event, compensation boundary event, and compensation handler. For the test, it would make it more clear that the compensation handler is triggered by the compensation throw event.

Reply:
:wrench: We should use proper element ids, for example, "undo-a". With the generated element ids, it is not clear that these belong to the compensation handler.

Reply:
>🔧 The limiting of the record stream doesn't work. The process instance is not completed yet. As a result, the test waits 5 seconds before it verifies the records.

I was aware of that, how we can limit the record stream in this case ? The limitation should work after #15066

Reply:
For now, we could replace: